mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/AbstractMessageMapper.java

```diff
@@ -101,7 +101,9 @@ public long getHighestModSeq(Mailbox mailbox) throws MailboxException {
 
     @Override
     public MessageMetaData add(Mailbox mailbox, MailboxMessage message) throws MailboxException {
-        message.setUid(uidProvider.nextUid(mailboxSession, mailbox));
+        MessageUid messageUid = uidProvider.nextUid(mailboxSession, mailbox);
+        message.setUid(messageUid);
+        message.setMessageId(messageIdProvider.from(mailbox.getMailboxId(), messageUid));

```

Comment:
that's too late. You should call an factory to create the Message so that the right backend puts an id into it earlier.
We should never pass around a message without a messageId